plasmapy/formulary/ionization.py

```diff
@@ -88,3 +88,84 @@ def ionization_balance(n: u.m ** -3, T_e: u.K) -> u.dimensionless_unscaled:
 
 Z_bal = ionization_balance
 """alias for :func:`ionization_balance`"""
+
+
+@validate_quantities(
+    T_e={"can_be_negative": False, "equivalencies": u.temperature_energy()}
+)
+def Saha(g_j, g_k, n_e: u.m ** -3, E_jk: u.J, T_e: u.K) -> u.dimensionless_unscaled:
+    r"""
+    The Saha equation, derived in statistical mechanics, gives an approximation of
+    the ratio of population of ions in two different ionization states in a plasma.
+    This approximation applies to plasmas in thermodynamic equilibrium where ionization
+    and recombination of ions with electrons are balanced.
+
+    .. math::
+        \frac{N_j}{N_k} = \frac{1}{n_e} \frac{g_j}{4 g_k a_0^{3}} \left( \frac{k_B T_e}{\pi E_H} \right)^{\frac{3}{2}}
+         \exp\left( \frac{-E_{jk}}{k_B T_e} \right)
+
+    Where :math:`k_B` is the Boltzmann constant,
+    :math:`a_0` is the Bohr radius,
+    :math:`E_H` is the ionization energy of Hydrogen,
+    :math:`N_j` and :math:`N_k` are the population of ions in the j and k states respectively.
+
+    Parameters
+    ----------
+    T_e : `~astropy.units.Quantity`
+        The electron temperature.
+
+    g_j : int
+        The degeneracy of the 'j'th ionization state.
+
+    g_k : int
+        The degeneracy of the 'k'th ionization state.
+
+    E_jk : `~astropy.units.Quantity`
+        The energy difference between ionization states j and k.
+
+    n_e : `~astropy.units.Quantity`
+        The electron number density of the plasma.
+
+    Warns
+    -----
+    ~astropy.units.UnitsWarning
+        If units are not provided, SI units are assumed.
+
+    Raises
+    ------
+    TypeError
+        The `T_e`, `E_jk`, or `n_e` is not a `~astropy.units.Quantity` and cannot be
+        converted into a ~astropy.units.Quantity.
+
+    ~astropy.units.UnitConversionError
+        If the `T_e`, `E_jk`, or `n` not in appropriate units.
+
+    Returns
+    -------
+    ratio: `~astropy.units.Quantity`
+        The ratio of population of ions in ionization state j to state k.
+
+    Examples
+    --------
+    >>> import astropy.units as u
+    >>> T_e = 5000 * u.K
+    >>> n = 1e19 * u.m ** -3
+    >>> g_j = 2
+    >>> g_k = 2
+    >>> E_jk = 1 * u.Ry
+    >>> Saha(g_j, g_k, n, E_jk, T_e)
+    <Quantity 3.29921072e-06>
+    >>> T_e = 1 * u.Ry
+    >>> n = 1e23 * u.m ** -3
+    >>> Saha(g_j, g_k, n, E_jk, T_e)
+    <Quantity 1114595.58601976>
+
+    """
+    E_h = 1 * u.Ry
+    A = g_j / (4 * g_k * a0 ** 3)
+    B = (k_B * T_e / (pi * E_h)) ** (3 / 2)
+    C = exp(-E_jk / (k_B * T_e)) / n_e
```

Comment:
Looking at this takes me back to `transport`... better names here, maybe? Do `degeneracy_factor, physical_constants, boltzmann_factor` sound good?

I just assume here the first one is related to degeneracies somehow, intuitively it rings that bell in my head.

Reply:
That makes sense, I didn't know how to rename them but that does seem good